### PR TITLE
Core/PacketIO: SMSG_PLAY_SOUND

### PR DESCRIPTION
**Pull request by Golrag**

### CODE REVIEW

**Comment by DDuarte** [comment]
Thanks

Why `SPlaySound` instead of simply `PlaySound`?

**Comment by Golrag** [comment]
Wanted to avoid confusion but can't remember what was confusing... Will change it back to PlaySound :p

File: src/server/game/Battlegrounds/Battleground.cpp [hunk 1]
```diff
@@ -667,22 +668,24 @@ void Battleground::SendChatMessage(Creature* source, uint8 textId, WorldObject*
...
-void Battleground::PlaySoundToAll(uint32 SoundID)
+void Battleground::PlaySoundToAll(uint32 soundID)
 {
-    WorldPacket data;
-    sBattlegroundMgr->BuildPlaySoundPacket(&data, SoundID);
-    SendPacketToAll(&data);
+    WorldPackets::Misc::PlaySound playSound;
+    sBattlegroundMgr->BuildPlaySoundPacket(&playSound, soundID);
+    SendPacketToAll(playSound.Write());
 }
```
Comment (DDuarte) [inline src/server/game/Battlegrounds/Battleground.cpp]: I'd suggest drop the BattlegroundMgr::BuildPlaySoundPacket and use the constructor of PlaySound that you already added here:

```c++
    WorldPackets::Misc::PlaySound playSound(ObjectGuid::Empty, soundID);
    SendPacketToAll(playSound.Write());
```

Comment (Golrag) [inline src/server/game/Battlegrounds/Battleground.cpp]: Done ! :)

**Pull request merged**